include/jemalloc/internal/ticker.h

```diff
@@ -32,14 +32,50 @@ ticker_read(const ticker_t *ticker) {
 	return ticker->tick;
 }
 
+/*
+ * Not intended to be a public API.  Unfortunately, on x86, neither gcc nor
+ * clang seems smart enough to turn
+ *   ticker->tick -= nticks;
+ *   if (unlikely(ticker->tick < 0)) {
+ *     fixup ticker
+ *     return true;
+ *   }
+ *   return false;
+ * into
+ *   subq %nticks_reg, (%ticker_reg)
+ *   js fixup ticker
+ *
+ * unless we force "fixup ticker" out of line.  In that case, gcc gets it right,
+ * but clang now does worse than before.  So, on x86 with gcc, we force it out
+ * of line, but otherwise let the inlining occur.  Ordinarily this wouldn't be
+ * worth the hassle, but this is on the fast path of both malloc and free (via
+ * tcache_event).
+ */
+#if defined(__GNUC__) && !defined(__clang__)				\
+    && (defined(__x86_64__) || defined(__i386__))
+/*
+ * -Wattributes complains about noinline inline functions, even though that's
+ *  not always incorrect (noinline being about optimization, and "inline" being
+ *  about symbol conflicts).  We'll disable -Wattributes temporarily.  Ideally,
+ *  we'd pragma push and pragma pop around the definition, but we need to
+ *  compile on GCC versions that don't support it.
+ */
+#pragma GCC diagnostic ignored "-Wattributes"
```

Comment:
Maybe something like  
```
static JEMALLOC_NOINLINE

#else 

static inline bool

#endif 
```
to avoid the Wattributes hassle?

Reply:
Hah. Ordinarily we don't do just plain static because the compiler will *also* complain about an unused static function. Here it's fine, but muscle memory is strong. I'll change it.